### PR TITLE
PROC-1346: Handle Unitless Allocation Edge case

### PR DESCRIPTION
**Pull request by zacharygoodwin**
Case where a test has multiple rules with missingExperimentalUnit and !missingExperimentalUnit and null identifier was being not handled correctly due to misunderstanding on null identifier case.

### CODE REVIEW

File: proctor-common/src/main/java/com/indeed/proctor/common/Proctor.java [hunk 1]
```diff
@@ -378,13 +378,7 @@ public ProctorResult determineTestGroups(
                     containsUnitlessAllocations ? VALUE_EXPRESSION_TRUE : VALUE_EXPRESSION_FALSE);
             final TestType testType = testChooser.getTestDefinition().getTestType();
             if (testChooser instanceof UnitlessTestChooser) {
-                identifier =
```
Comment (jamesallen84) [inline proctor-common/src/main/java/com/indeed/proctor/common/Proctor.java:381]: Why the change here?

**Review by jamesallen84 — APPROVED** [review]

File: proctor-common/src/main/java/com/indeed/proctor/common/Proctor.java (hunk 1 above)
Comment (zacharygoodwin) [reply proctor-common/src/main/java/com/indeed/proctor/common/Proctor.java:381]: After further discussion with steven, determined that null should be rejected and not able to get a result. 

This was causing a minor bug for an edge case where a non empty result was being returned. This fixes the bug and cleans up the logic. Making it more clear null is being rejected in the UnitlessTestChooser.

The edge case was in testDetermineTestGroupsWithNullIdentifier_UnitlessEnabled_falseAndTrueInRule

File: proctor-common/src/test/java/com/indeed/proctor/common/TestProctor.java [hunk 2]
```diff
@@ -475,6 +476,155 @@ public void testDetermineTestGroupsWithInvalidIdentifier() {
...
+
+        assertThat(proctorResult.getBuckets()).isEmpty();
+        assertThat(proctorResult.getAllocations()).isEmpty();
+        assertThat(proctorResult.getTestDefinitions())
+                .containsOnlyKeys("X")
+                .containsEntry("X", testDefinitionX);
+    }
+
+    @Test
+    public void testDetermineTestGroupsWithNullIdentifier_UnitlessEnabled_falseAndTrueInRule() {
```
Comment (stevenchen-indeed) [inline proctor-common/src/test/java/com/indeed/proctor/common/TestProctor.java:576]: do we have a test case where the test has both missingExperimentalUnit and !missingExperimentalUnit, just like in this test, but the identifier is ""?

Comment (zacharygoodwin) [reply proctor-common/src/test/java/com/indeed/proctor/common/TestProctor.java:576]: Added that case + valid identifier case
